### PR TITLE
Resolve 244: Built-in Annotation Literal Tests

### PR DESCRIPTION
**Pull request by neilgsyoung**

### CODE REVIEW

**Comment by neilgsyoung** [comment]
#build

File: dev/com.ibm.ws.cdi.2.0_fat/publish/servers/cdi20BuiltinAnnoServer/server.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<server description="new server">
+
+	<featureManager>
+	    <feature>osgiconsole-1.0</feature>
+        <feature>cdi-2.0</feature>
+        <feature>jndi-1.0</feature>
+        <feature>jsp-2.3</feature>
+        <feature>servlet-4.0</feature>
+        <feature>componentTest-1.0</feature>
+        <feature>jndi-1.0</feature>
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/publish/servers/cdi20BuiltinAnnoServer/server.xml]: Is this feature needed?

File: dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Battenberg.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@Named("BATTENBERG")
+public class Battenberg implements Cake, Serializable {
+
+    /**  */
+    private static final long serialVersionUID = 1L;
+
+    @Inject
+    private SomeRandomClass someRandom;
+
+    public Battenberg() {};
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Battenberg.java]: Is this needed?

File: dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Battenberg.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see builtinAnnoApp.web.BuiltinAnnoType#greeting()
+     */
+    @Override
+    public String greeting() {
+        return "Hello Builtin Session Bean named BATTENBERG!";
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Battenberg.java]: This is minor. It is not a session bean. It is a CDI bean.

File: dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Rumbaba.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@ApplicationScoped
+@Named("RUMBABA")
+public class Rumbaba implements Cake {
+    @Inject
+    private SomeRandomClass someRandom;
+
+    public Rumbaba() {};
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/Rumbaba.java]: not needed, I think

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

File: dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/BuiltinAnnoServlet.java [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+    /**
+     * Test the javax.enterprise.util.Nonbinding.Literal class introduced in CDI2.0.
+     *
+     * The testing mechanism involves the use of a CDI Extension in the CakeExtension class which observes the
+     * Lemondrizzle class through a ProcessAnnotatedType event. When the event fires, as Weld processes the Lemondrizzle bean,
+     * the CakeExtension programmatically adds a @Typed(Cake.class) annotation to the Lemondrizzle bean using a
+     * Typed.Literal. We check that the Lemondrizzle can (thanks to this annotation) NOT be found by searching for a Lemondrizzle
+     * bean but CAN be found by a generic Cake search.
+     */
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/test-applications/builtinAnnoLiteralsApp/src/builtinAnnoApp/web/BuiltinAnnoServlet.java]: The comments are for the rest two methods are in the wrong order.

**Comment by neilgsyoung** [comment]
#build

**Pull request merged**